docs/source/typevars.py

```diff
@@ -0,0 +1,103 @@
+"""Transform references to typevars to avoid missing reference errors.
+
+See https://github.com/sphinx-doc/sphinx/issues/7722 also.
+"""
+from __future__ import annotations
+
+import re
+
+from pathlib import Path
+
+from sphinx.addnodes import pending_xref, Element
+from sphinx.application import Sphinx
+from sphinx.environment import BuildEnvironment
+from sphinx.errors import NoUri
+
+
+def identify_typevars() -> None:
+    """Record all typevars in trio."""
+    trio_folder = Path(__file__, "..", "..", "..", "trio").resolve()
+    print(trio_folder)
+    for filename in trio_folder.rglob("*.py"):
+        with open(filename, encoding="utf8") as f:
+            for line in f:
+                # A simple regex should be sufficient to find them all, no need to actually parse.
+                match = re.search(
+                    r"^\s*(\w+)\s*=\s*(TypeVar|TypeVarTuple|ParamSpec)\(",
```

Comment:
it might be easier to use the string passed into the typevar

Reply:
Perhaps it might, but since this is anchored at the start of the string, it'll quickly fail for any non-assignment lines.